internal/scans/scan_progress.go

```diff
@@ -0,0 +1,100 @@
+package scans
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/rs/zerolog/log"
+)
+
+const timeout = 5 * time.Second
+
+// ScanProgress is a type that's used for tracking current running scans, listen to their progress events
+// and invoke cancellations or "done" signals. This allows throttling or cancelling previous scans instead of
+// having many scans running at once for the same files.
+//
+// The correct usage would be to create a NewScanProgress and call "go Listen()" to
+// listen for cancel/done signals in the background.
+// ScanProgress operations should be done in locked sections.
+type ScanProgress struct {
```

Comment:
Not for today, but would be nice to integrate a progress tracker here, as we have some overlap in the functionality :)